src/launchpad/utils/treemap_builder.py

```diff
@@ -405,17 +401,15 @@ def _calculate_category_breakdown(self, file_analysis: FileAnalysis) -> Dict[str
         """Calculate size breakdown by category."""
         breakdown: Dict[str, Dict[str, int]] = defaultdict(lambda: {"install": 0, "download": 0})
 
-        for files in file_analysis.files_by_type.values():
-            for file_info in files:
-                treemap_type = file_info.treemap_type
-                category = treemap_type.value
+        for file_info in file_analysis.files:
+            treemap_type = file_info.treemap_type
 
-                # Use filesystem block-aligned size for install calculations
-                install_size = self._calculate_aligned_install_size(file_info)
-                download_size = int(install_size * self.download_compression_ratio)
+            # Use filesystem block-aligned size for install calculations
+            install_size = self._calculate_aligned_install_size(file_info)
+            download_size = int(install_size * self.download_compression_ratio)
 
-                breakdown[category]["install"] += install_size
-                breakdown[category]["download"] += download_size
+            breakdown[treemap_type]["install"] += install_size
```

Comment:
Assuming this is also intentional? Feels a bit unrelated to this change, maybe better to move into a stacked PR?